<launch>
  <!-- Arguments -->
  <arg name="robot_name"    default=""/>
  <arg name="robot_tf"       default=""/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id"   default="map"/>
  <arg name="odom_topic" default="/odom" />
  <arg name="laser_topic" default="/scan" />
  
  <arg name="move_forward_only" default="false"/>

  <!-- move_base -->
  <group ns="$(arg robot_name)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
      <rosparam file="$(find robot)/param/costmap_common_params_robot2.yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find robot)/param/costmap_common_params_robot2.yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find robot)/param/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find robot)/param/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find robot)/param/move_base_params.yaml" command="load" />
      <rosparam file="$(find robot)/param/dwa_local_planner_params_robot2.yaml" command="load" />
      
      <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
      <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
      <param name="global_costmap/scan/sensor_frame" value="$(arg robot_tf)base_scan"/>
      <param name="global_costmap/scan/topic" value="$(arg laser_topic)"/>
      
      <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
      <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
      <param name="local_costmap/scan/sensor_frame" value="$(arg robot_tf)base_scan"/>
      <param name="local_costmap/scan/topic" value="$(arg laser_topic)"/>
      
      <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>
      
      <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
      <remap from="odom"  to="$(arg odom_topic)"/>
      <remap from="map"   to="/map"/>
      <remap from="scan" to="$(arg laser_topic)"/>
      <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    </node>
  </group>
</launch>
